### PR TITLE
Virtual issues

### PR DESCRIPTION
**Pull request by waywardmonkeys**

### CODE REVIEW

File: lib/Target/JSBackend/JSTargetTransformInfo.cpp [hunk 1]
```diff
@@ -55,24 +55,26 @@ class JSTTI : public ImmutablePass, public TargetTransformInfo {
...
+                                  OperandValueProperties Opd1PropInfo = OP_None,
+                                  OperandValueProperties Opd2PropInfo = OP_None) const override;
 
-  virtual unsigned getVectorInstrCost(unsigned Opcode, Type *Val,
-                                      unsigned Index = -1) const;
+  unsigned getVectorInstrCost(unsigned Opcode, Type *Val,
+                              unsigned Index = -1) const override;
 
-  virtual void getUnrollingPreferences(Loop *L, UnrollingPreferences &UP) const;
+  void getUnrollingPreferences(const Function *F, Loop *L, UnrollingPreferences &UP) const override;
```
Comment (kripken) [inline lib/Target/JSBackend/JSTargetTransformInfo.cpp]: this looks like the right signature in the headers, but i wonder how it worked without this change, so i seem to be misunderstanding something

cc @sunfishcode

Comment (sunfishcode) [inline lib/Target/JSBackend/JSTargetTransformInfo.cpp]: It looks like the signature in the base class changed in LLVM 3.6.

Comment (kripken) [inline lib/Target/JSBackend/JSTargetTransformInfo.cpp]: yes, but how did this work without @waywardmonkeys 's fix? was it overloading the method name, but the method wasn't actually being called? if so it sounds like something we should have tests for.

Comment (waywardmonkeys) [inline lib/Target/JSBackend/JSTargetTransformInfo.cpp]: The compiler was emitting warnings for both of the changed signatures. My change to use ``override`` makes them into compilation errors.

Comment (sunfishcode) [inline lib/Target/JSBackend/JSTargetTransformInfo.cpp]: Yes, I assume it wasn't being called. These are heuristics, so they aren't strictly necessary. Also, yes, we should ideally have tests for these. The code that's there right now is just some initial guesswork, something we can start experimenting with.

**Pull request merged**